File: frontend/src/dto/auction/updateAuctionDto.js
import Joi from "joi";

class UpdateAuctionDto {
    title = "";
    description = "";
    endTime = "";
    auctionPicturePath  = "";
}

export function validateUpdateAuctionDto(updateAuctionDto) {
    const schema = Joi.object({
        title: Joi.string().optional(),
        description: Joi.string().optional(),
        endTime: Joi.date().iso().optional(),
        auctionPicturePath: Joi.string().optional()
    });

    return schema.validate(updateAuctionDto);
}
export default UpdateAuctionDto;

File: backend/DTO/Auction/UpdateAuctionDto.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace DreamBid.Dtos.Auction
{
    public class UpdateAuctionDto
    {
        [Required]
        [StringLength(100)]
        public string Title { get; set; }

        [Required]
        public string Description { get; set; }
        public DateTime EndTime { get; set; }
        

    }
}